gossip/emitter/proposals.go

```diff
@@ -181,27 +198,34 @@ func makeProposal(
 	currentFrame idx.Frame,
 	transactionScheduler txScheduler,
 	candidates scheduler.PrioritizedTransactions,
+	randaoMixer randaoMixer,
 	durationMetric timerMetric,
 	timeoutMetric counterMetric,
-) *inter.Proposal {
+) (*inter.Proposal, error) {
 	// Compute the gas limit for the next block. This is the time since the
 	// previous block times the targeted network throughput.
 	lastBlockTime := latestBlock.Time
 	if lastBlockTime >= newBlockTime {
 		// no time has passed, so no new proposal can be made
-		return nil
+		return nil, nil
 	}
 	effectiveGasLimit := getEffectiveGasLimit(
 		newBlockTime.Time().Sub(lastBlockTime.Time()),
 		rules.Economy.ShortGasPower.AllocPerSec, // TODO: consider using a new rule set parameter
 	)
 
+	randaoReveal, randaoMix, err := randaoMixer.MixRandao(latestBlock.PrevRandao)
+	if err != nil {
+		// If the randao mixer fails, we cannot create a proposal.
+		return nil, ErrRandaoGenerationFailed
```

Comment:
I think the actual `err` should be forwarded to provide additional details on the cause, in addition to potential context information you can add here. Something like
```suggestion
		return nil, fmt.Errorf("randao reveal generation failed: %w", err)
```
Maybe?